codegen/src/main/java/software/amazon/awssdk/codegen/internal/Utils.java

```diff
@@ -307,6 +309,10 @@ public static ShapeModel findShapeModelByC2jNameIfExists(IntermediateModel inter
         return null;
     }
 
+    public static List<ShapeModel> findShapesByC2jName(IntermediateModel intermediateModel, String shapeC2jName) {
+        return intermediateModel.getShapes().values().stream().filter(s -> s.getC2jName().equals(shapeC2jName)).collect(toList());
```

Comment:
This had me stumped for a bit, but for the operation level request/response shapes we actually create a new `ShapeModel` but leave the original c2j shape in the intermediate model list. So when It was originally doing `findShapeByC2JName` it just brought back the first one it found and not the `Response` shape that we'd created.